clients/privacy-center/cypress/e2e/consent-banner.cy.ts

```diff
@@ -188,11 +215,44 @@ describe("Consent banner", () => {
           });
           cy.getByTestId("toggle-Essential").click();
 
-          // DEFER: intercept and check the API call once it is hooked up
           cy.getByTestId("Save-btn").click();
           // Modal should close after saving
           cy.getByTestId("consent-modal").should("not.exist");
 
+          // check that consent was sent to Fides API
+          cy.wait("@patchPrivacyPreference").then((interception) => {
+            const { body } = interception.request;
+            const expected = {
+              // browser_identity.fides_user_device_id is intentionally left out here
+              // so we can later assert to be any string
+              preferences: [
+                {
+                  privacy_notice_history_id:
+                    "pri_b09058a7-9f54-4360-8da5-4521e8975d4f",
+                  preference: "opt_in",
+                },
+                {
+                  privacy_notice_history_id:
+                    "pri_b09058a7-9f54-4360-8da5-4521e8975d4e",
+                  preference: "acknowledge",
+                },
+              ],
+              privacy_experience_history_id: "2342345",
+              user_geography: "us_ca",
+              method: ConsentMethod.button,
+            };
+            // uuid is generated automatically if the user has no saved consent cookie
+            expect(body.browser_identity.fides_user_device_id).to.be.a(
+              "string"
+            );
+            expect(body.preferences).to.eql(expected.preferences);
+            expect(body.privacy_experience_history_id).to.eql(
+              expected.privacy_experience_history_id
+            );
+            expect(body.user_geography).to.eql(expected.user_geography);
+            expect(body.method).to.eql(expected.method);
+          });
+
           // check that the cookie updated
           cy.waitUntilCookieExists(CONSENT_COOKIE_NAME).then(() => {
             cy.getCookie(CONSENT_COOKIE_NAME).then((cookie) => {
```

Comment:
might be worth adding a check that the `fides_user_device_id` saved to the cookie is the one that was in the request